BraveRewardsTests/LedgerDatabaseTests.mm

```diff
@@ -144,6 +147,638 @@ - (void)testNonZeroExcludedCount
+{
+  const auto probi = @"10000000000000000";
+  const BATRewardsCategory category = BATRewardsCategoryOneTimeTip;
+  const auto now = [[NSDate date] timeIntervalSince1970];
+  
+  [self backgroundSaveAndWaitForExpectation:^{
+    [BATLedgerDatabase insertContributionInfo:probi month:month year:year date:now
+                                 publisherKey:publisherId category:category];
+  }];
+}
+
+- (void)makeOneTimeTip:(NSString *)publisherId month:(const int)month year:(const int)year
+                 probi:(NSString *)probi category:(BATRewardsCategory)category
+{
+  const auto now = [[NSDate date] timeIntervalSince1970];
+  
+  [self backgroundSaveAndWaitForExpectation:^{
+    [BATLedgerDatabase insertContributionInfo:probi month:month year:year date:now
+                                 publisherKey:publisherId category:category];
+  }];
+}
+
+#pragma mark - Activity Info
+
+- (void)testInsertOrUpdateActivityInfoFromPublisherEmptyId
+{
+  const auto info = [self createBATPublisherInfo:@"111" reconcileStamp:111 percent:11 createActivityInfo:NO];
+  XCTAssert([self countMustBeEqualTo:0 forEntityName:@"ActivityInfo"]);
+  
+  info.id = @"";
+  [self waitForFailedSaveAttempt:^{
+    [BATLedgerDatabase insertOrUpdateActivityInfoFromPublisher:info];
+  }];
+  XCTAssert([self countMustBeEqualTo:0 forEntityName:@"ActivityInfo"]);
+}
+
+- (void)testInsertOrUpdateActivitiesInfoFromPublishers
+{
+  const auto p1 = [self createBATPublisherInfo:@"111" reconcileStamp:111 percent:11 createActivityInfo:NO];
+  const auto p2 = [self createBATPublisherInfo:@"222" reconcileStamp:222 percent:22 createActivityInfo:NO];
+  const auto p3 = [self createBATPublisherInfo:@"333" reconcileStamp:333 percent:33 createActivityInfo:NO];
+  
+  const auto publishers = @[p1, p2, p3];
+  
+  XCTAssert([self countMustBeEqualTo:0 forEntityName:@"ActivityInfo"]);
+  
+  [self backgroundSaveAndWaitForExpectation:^{
+    [BATLedgerDatabase insertOrUpdateActivitiesInfoFromPublishers:publishers];
+  }];
+  
+  XCTAssert([self countMustBeEqualTo:3 forEntityName:@"ActivityInfo"]);
+}
+
+- (void)testDeleteActivityInfoWithPublisherID
+{
+  [self createBATPublisherInfo:@"1" reconcileStamp:10 percent:11 createActivityInfo:YES];
+  [self createBATPublisherInfo:@"2" reconcileStamp:10
+                                               percent:22 createActivityInfo:YES];
+  
+  XCTAssert([self countMustBeEqualTo:2 forEntityName:@"ActivityInfo"]);
+  
+  [self backgroundSaveAndWaitForExpectation:^{
+    [BATLedgerDatabase deleteActivityInfoWithPublisherID:@"1" reconcileStamp:10];
+  }];
+  XCTAssert([self countMustBeEqualTo:1 forEntityName:@"ActivityInfo"]);
+}
+
+#pragma mark - publishersWithActivityFromOffset
+
+- (void)testPublishersWithFiltersEmptyFilter
+{
+  const auto offset = 0;
+  const auto limit = 0;
+  const auto filter = [[BATActivityInfoFilter alloc] init];
+  
+  [self createBATPublisherInfo:@"1" reconcileStamp:10 percent:10 createActivityInfo:true];
```

Comment:
[Minor/Lint]: I see you started using `true` in some places, and `YES` in others. It doesn't matter since they will compile to the same thing, but for consistency sake I'd rather we stick with one. I've used `YES` throughout, out of Obj-C habit, but if you want to use `true` instead to match Swift for this file thats fine by me.